lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -954,9 +960,40 @@ instance FromJSON ApiAddressDerivationPath where
     parseJSON = fmap ApiAddressDerivationPath . parseJSON
 
 instance ToJSON ApiAddressDerivationSegment where
-    toJSON = genericToJSON defaultRecordTypeOptions
+    toJSON (ApiAddressDerivationSegment (ApiRelativeAddressIndex ix) typ)
+        | typ == Hardened = toJSON (show ix <> "H")
```

Comment:
`show` makes me feel uneasy, since it's for debugging purposes mostly, not serialisation, although this use case seems safe.

There's a more elaborate way via Text: https://github.com/haskell/text/issues/218#issue-296457899
```hs
T.toStrict . B.toLazyText . B.decimal $ ix
```

Reply:
Yes, for Int / Word type, show is rather safe so I usually don't bother. The result is the same and quite more readable than more elaborated ways.

Reply:
I think we could also just go for `toText` here since we have such instances for Word / Int I believe. Adding them otherwise would be nice.